constraint/pkg/types/validation.go

```diff
@@ -65,7 +65,15 @@ func (r *Response) TraceDump() string {
 	b := &strings.Builder{}
 	_, _ = fmt.Fprintf(b, "Target: %s\n", r.Target)
 	if r.Trace == nil {
-		_, _ = fmt.Fprintf(b, "Trace: TRACING DISABLED\n\n")
+		if r.Results != nil {
+			// only say "Trace: TRACING DISABLED" if there are results
+			// otherwise, we risk to confuse consumers who see the msg
+			// and think that evaluation did not happen.
+
+			// Note that if there were NO violating results AND the trace
+			// was turned on, then r.Trace != nil.
+			_, _ = fmt.Fprintf(b, "Trace: TRACING DISABLED\n\n")
```

Comment:
nit: make this a const that you can also test in unit test?

Reply:
added in!